core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -193,13 +193,6 @@ private[spark] class TaskSchedulerImpl private[scheduler](
       val stageTaskSets =
         taskSetsByStageIdAndAttempt.getOrElseUpdate(stage, new HashMap[Int, TaskSetManager])
       stageTaskSets(taskSet.stageAttemptId) = manager
-      val conflictingTaskSet = stageTaskSets.exists { case (_, ts) =>
```

Comment:
actually, that is not really the point of this check.  Its just checking if one stage has two tasksets (aka stage attempts), where both are in the "non-zombie" state.  It doesn't do any checks at all on what tasks are actually in those tasksets.

This is just checking an invariant which we believe to always be true, but we figure its better to fail-fast if we hit this condition, rather than proceed with some inconsistent state.  This check was added because behavior gets *really* confusing when the invariant is violated, and though we think it should always be true, we've still hit cases where it happens.